docs/content/Examples-Tutorials-Recipes/Column-based-access.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Column-Based Access
+permalink: /recipes/column-based-access
+category: Examples & Tutorials
+subCategory: Access control
+menuOrder: 2
+---
+
+## Use case
+
+We want to manage user access to different data depending on a database
+relationship. In the recipe below, we will manage supplier access to their
+products. A supplier can't see other supplier's products.
+
+## Data schema
+
+To implement column-based access, we will use supplier's email from
+a [JSON Web Token](https://cube.dev/docs/security), and the
+[`queryRewrite`](https://cube.dev/docs/security/context#using-query-rewrite)
+extension point to manage data access.
+
+We have `Products` and `Suppliers` tables with `hasOne` relationship from
+products to suppliers:
+
+```javascript
+// schema/Products.js
+
+cube(`Products`, {
+  sql: `SELECT * FROM public.products`,
+
+  joins: {
+    Suppliers: {
+      sql: `${Suppliers}.id = ${CUBE}.supplier_id`,
+      relationship: `hasOne`
+    }
+  },
+  
+  dimensions: {
+    name: {
+      sql: `name`,
+      type: `string`
+    }
+  }
+});
+```
+```javascript
+// schema/Suppliers.js
+
+cube(`Suppliers`, {
+  sql: `SELECT * FROM public.suppliers`,
+
+  dimensions: {
+    email: {
+      sql: `email`,
+      type: `string`
+    }
+  }
+});
+```
+
+## Configuration
+
+Let's add the supplier email filter if a query includes any dimension from the `Products` cube:
+
+```javascript
+module.exports = {
+  queryRewrite: (query, { securityContext }) => {
+    const queryDimensions = Array.from(
+      query.dimensions,
+      (element) => element.split('.')[0]
+    );
+
+    if (queryDimensions.includes('Products')) {
+      query.filters.push({
+        member: `Suppliers.email`,
+        operator: 'equals',
+        values: [securityContext.email],
```

Comment:
Also, I think it's great to check if the email is present, fail the query otherwise.